content/get-haiku/r1beta4/_index.md

```diff
@@ -0,0 +1,165 @@
+    </tr>
+</tbody>
+</table>
+</div>
+
+### Torrents
+
+<p>In order to assist with distribution, we have also created .torrent files that can be used to download and seed the release files for others.<br/>
+<b>Please continue seeding the torrents if you can!</b></p>
+<ul>
+ <li><a class="track" href="https://s3.us-east-1.wasabisys.com/haiku-release/r1beta4/haiku-r1beta4-x86_gcc2h.torrent">32-bit Torrent</a></li>
+ <li><a class="track" href="https://s3.us-east-1.wasabisys.com/haiku-release/r1beta4/haiku-r1beta4-x86_64.torrent">64-bit Torrent</a></li>
+</ul>
+
+### Checksums
+
+<p>To make sure that you have got the right file and that it has not been corrupted during file transfer, it is always wise to ensure the integrity of your download. To that end, you can use the SHA256 checksums shown below to verify that what you have is the correct file.</p>
+
+<pre>
+6e4b33045a0c36efb2b8fe758ef3550bf87298a0c5a7c2fbdf4b155e19397ac7  haiku-r1beta4-x86_64-anyboot.iso
+c184e609a6c0021f9acd2df9fc111c2e12d01107d01f943fbea66c936b063f88  haiku-r1beta4-x86_gcc2h-anyboot.iso
+</pre>
+
+<p>
+	If you are in Haiku, macOS or Linux, you can use the <code>sha256sum</code>
+	command from the Terminal to generate a checksum for your downloaded
+	file and verify against those above. If you are in Windows, you can use
+	a tool such as <a href="https://implbits.com/products/hashtab/" title="Download hashtab">hashtab</a> to display checksums for files directly in explorer.
+</p>
+
+### Cryptographic Signatures
+
+<p>
+To further validate releases, you can also check the cryptographic signature of the release media against our public key using <a href="https://jedisct1.github.io/minisign/">Minisign</a> (available for Haiku, Windows, Linux, BSD, and OS X). The matching .minisig file needs to be placed next to the iso file on your filesystem.
+</p>
+
+<ul>
+  <li><a class="track" href="https://s3.us-east-1.wasabisys.com/haiku-release/r1beta4/haiku-r1beta4-x86_gcc2h-anyboot.iso.minisig">32-bit minisig</a></li>
+  <li><a class="track" href="https://s3.us-east-1.wasabisys.com/haiku-release/r1beta4/haiku-r1beta4-x86_64-anyboot.iso.minisig">64-bit minisig</a></li>
+</ul>
+
+{{< alert-info "Haiku signing key" "Haiku's public signing key is also available at /boot/system/data/trust_db/haiku-2019.pub under recent versions of Haiku." >}}
+
+```shell script
+$ ls
+haiku-r1beta4-x86_64-anyboot.iso
+haiku-r1beta4-x86_64-anyboot.iso.minisig
+
+$ minisign -Vm haiku-r1beta4-x86_64-anyboot.iso -P RWTPfbfFRi5b+T02aw733p5dp/UI/1Z946YWBInmkPmBlx2Zy1Vyw9kP
+Signature and comment signature verified
+Trusted comment: timestamp:1671733559	file:haiku-r1beta4-x86_64-anyboot.iso	hashed
+```
+
+## Upgrading from R1/beta3
+
+There is an upgrade path available for users that are currently using Haiku R1/beta3. If you want to do this upgrade, you will have to use the Terminal (not SoftwareUpdater!) to issue some commmands. To do so,
```

Comment:
Is the warning about SoftwareUpdater still applicable? There were problems with it in previeous releases but I thinè they are fixed?

Reply:
You need to do a full-sync to successfully upgrade to R1/Beta4.   Software updater doesn't do that still it seems?

Reply:
It does, actually: https://xref.landonf.org/source/xref/haiku/src/apps/softwareupdater/SoftwareUpdaterApp.cpp#50

However, there are two reasons to stick with Terminal. One is that you have to change the repos, and this is something SoftwareUpdater can't do (and using Repositories preferences is a bit more cumbersome than Terminal here.) The other is that if something goes wrong it is much easier to just copy-and-paste whatever the error message was in `pkgman`. Perhaps these can be worked on for a future release.

I can at least remove the "(not SoftwareUpdater)" however.